node/hubController.js

```diff
@@ -0,0 +1,211 @@
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            return this._createTranslator(hubInfo.translator, authInfo).then((hubInstance) => {
+                // platform get
+                var deviceInfo = {};
+                deviceInfo.hub = hubInstance;
+                deviceInfo.deviceInfo = {};
+                deviceInfo.deviceInfo.opent2t = opent2tBlob;
+                return this._getProperty(opent2tBlob.translator, deviceInfo, "get", true);
+            });
+        }).catch((err) => {
+            this._logError(err, "getPlatform");
+        });
+    }
+
+    /**
+     * given the specific hub id, opent2tblob, and resourceId, sets it with the given resourceBlob
+     */
+    setResource(hubId, authInfo, opent2tBlob, deviceId, resoureceId, resourceBlob) {
+        console.log("----------------- setResource");
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            return this._createTranslator(hubInfo.translator, authInfo).then((hubInstance) => {
+                // resource set
+                var deviceInfo = {};
+                deviceInfo.hub = hubInstance;
+                deviceInfo.deviceInfo = {};
+                deviceInfo.deviceInfo.opent2t = opent2tBlob;
+
+                var propertyName = "postDevices" + this._capitalizeFirstLetter(resoureceId);
+
+                return this._setProperty(opent2tBlob.translator, deviceInfo, propertyName, deviceId, resourceBlob);
+            });
+        }).catch((err) => {
+            this._logError(err, "setResource");
+        });
+    }
+
+
+    /// *********************************
+    /// helper methods
+    /// *********************************
+    _getHubInfo(hubId) {
+        console.log("----------------- _getHubInfo");
+        return this.supportedHubs().then((hubs) => {
+            // find the hub referenced by hubId
+            var hubInfo = undefined;
+            for (var i = 0; hubInfo == undefined && i < hubs.length; i++) {
+                var hub = hubs[i];
+
+                // intentional ==
+                if (hub.id == hubId) {
+                    hubInfo = hub;
+                }
+            }
+
+            if (hubInfo == undefined) {
+                console.log("invalid hub id");
+                throw new Error("Invalid hub id");
+            }
+
+            return hubInfo;
+        });
+    }
+
+    _getProperty(translator, deviceInfo, property, value) {
+        console.log("----------------- _getProperty " + property);
+
+        if (typeof translator === "object") {
+            return this.OpenT2T.invokeMethodAsync(translator, "", property, [value]);
+        } 
+        else {
+            return this._createTranslator(translator, deviceInfo).then(translatorInstance => {
+                return this.OpenT2T.invokeMethodAsync(translatorInstance, "", property, [value]);
+            });
+        }
+    }
+    
+    _setProperty(translatorName, deviceInfo, property, deviceId, value) {
```

Comment:
they do different things, updated getProperty to be invokeMethod instead.

---
In reply to: [88942344](https://github.com/openT2T/cloud/pull/9#discussion_r88942344) [](ancestors = 88942344)